server/routes.go

```diff
@@ -120,6 +120,10 @@ func (s *Server) GenerateHandler(c *gin.Context) {
 	}
 
 	caps := []Capability{CapabilityCompletion}
+	if req.Tools != nil {
+		caps = append(caps, CapabilityTools)
```

Comment:
It would be useful to add a helpful error message here when the `tools` capability is not supported similar to when the completion capability is missing. This will be a pretty frequent error case as new models are updated to support tools and callers start using the API.

Consider adding a test for this case.

Reply:
`CheckCapabilities` returns an error when required capabilities are missing that is propogated up to the user as an api error, e.g.

```
$ curl -s 127.0.0.1:11434/api/chat -d '{
    "model": "llama3",
    "tools": [
        {
            "type":"function",
            "function":{}
        }
    ]
}'
{"error":"llama3 does not support tools"}
```

Reply:
Awesome, thank you, wanted to make sure it was readable.